indexer/processors/bridge/l1_bridge_processor.go

```diff
@@ -0,0 +1,249 @@
+package bridge
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	contracts "github.com/ethereum-optimism/optimism/indexer/processors/contracts_bedrock"
+	"github.com/ethereum-optimism/optimism/op-e2e/config"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// L1ProcessInitiatedBridgeEvents will query the database for new bridge events that have been initiated between
+// the specified block range. This covers every part of the multi-layered stack:
+//  1. OptimismPortal
+//  2. L1CrossDomainMessenger
+//  3. L1StandardBridge
+func L1ProcessInitiatedBridgeEvents(log log.Logger, db *database.DB, fromHeight *big.Int, toHeight *big.Int) error {
+	// (1) OptimismPortal
+	optimismPortalTxDeposits, err := contracts.OptimismPortalTransactionDepositEvents(config.L1Deployments.OptimismPortalProxy, db, fromHeight, toHeight)
+	if err != nil {
+		return err
+	}
+
+	ethDeposits := []*database.L1BridgeDeposit{}
+	portalDeposits := make(map[logKey]*contracts.OptimismPortalTransactionDepositEvent, len(optimismPortalTxDeposits))
+	transactionDeposits := make([]*database.L1TransactionDeposit, len(optimismPortalTxDeposits))
+	for i := range optimismPortalTxDeposits {
+		depositTx := optimismPortalTxDeposits[i]
+		portalDeposits[logKey{depositTx.Event.BlockHash, depositTx.Event.LogIndex}] = &depositTx
+		transactionDeposits[i] = &database.L1TransactionDeposit{
+			SourceHash:           depositTx.DepositTx.SourceHash,
+			L2TransactionHash:    types.NewTx(depositTx.DepositTx).Hash(),
+			InitiatedL1EventGUID: depositTx.Event.GUID,
+			GasLimit:             depositTx.GasLimit,
+			Tx:                   depositTx.Tx,
+		}
+
+		// catch ETH transfers to the portal contract.
+		if len(depositTx.DepositTx.Data) == 0 && depositTx.DepositTx.Value.BitLen() > 0 {
+			ethDeposits = append(ethDeposits, &database.L1BridgeDeposit{
+				TransactionSourceHash: depositTx.DepositTx.SourceHash,
+				BridgeTransfer:        database.BridgeTransfer{Tx: transactionDeposits[i].Tx, TokenPair: database.ETHTokenPair},
+			})
```

Comment:
nit: seems unnecessary to handle ethDeposits while processing we can figure out if something is an eth deposit or not purely via reading

Reply:
So the reason I had this is because the deposits handler reads from the bridge deposits table and not the transactions deposits table. By adding this here, it allows an entry to be created in the bridge deposits table making it show in the API.

Reply:
Alternatively, we can have the deposits/withdrawals return **all** unique deposits/withdrawals like arbitrary messages and our API can simply hydrate a deposit/withdrawal with optional properties
- transaction deposit X is a StandardBridge message and here's the token information 
- transaction deposit Y is a L1CrossDomain message and here's the target contract and message it'll execute 
- transaction deposit Z is solely a transaction deposit via the portal. here's the raw info

And in this case we can also have a optional query param to only return results of a given type. "i only want standardbridge deposits/withdrawals"